app/renderer/reducers/Session.js

```diff
@@ -336,6 +336,19 @@ export default function session (state = INITIAL_STATE, action) {
         ...omit(action.state, ['server']),
       };
 
+    case SET_STATE_FROM_SAVED:
+      if (!state.visibleProviders.includes(action.state.serverType) &&
+        action.state.serverType !== ServerTypes.local &&
+        action.state.serverType !== ServerTypes.remote
+      ) {
+        state.visibleProviders.push(action.state.serverType);
```

Comment:
do we want to check that the provider actually is valid?

Reply:
Good thinking. Done.